src/test/suite/connectionController.test.ts

```diff
@@ -1475,43 +1475,68 @@ suite('Connection Controller Test Suite', function () {
       assert.strictEqual(isConnectionChanged, true);
     });
 
-    test('should track and also notify the users of failed keytar secrets migration', async () => {
-      await testConnectionController.addNewConnectionStringAndConnect(
-        TEST_DATABASE_URI
-      );
-      const [savedConnection] = testConnectionController.getSavedConnections();
-
-      testConnectionController.clearAllConnections();
+    test('should track and also notify the users of unique failed keytar secrets migration (in the current load of extension)', async () => {
+      testSandbox.replace(testStorageController, 'get', (key, storage) => {
+        if (
+          storage === StorageLocation.WORKSPACE ||
+          key === StorageVariables.WORKSPACE_SAVED_CONNECTIONS
+        ) {
+          return {};
+        }
 
+        return {
+          'random-connection-1': {
+            id: 'random-connection-1',
+            name: 'localhost:27017',
+            storageLocation: 'GLOBAL',
+            secretStorageLocation: SecretStorageLocation.Keytar,
+            connectionOptions: {
+              connectionString:
+                'mongodb://localhost:27017/?readPreference=primary&ssl=false',
+            },
+          },
+          'random-connection-2': {
+            id: 'random-connection-2',
+            name: 'localhost:27018',
+            storageLocation: 'GLOBAL',
+            connectionOptions: {
+              connectionString:
+                'mongodb://localhost:27018/?readPreference=primary&ssl=false',
+            },
+          },
+        } as any;
+      });
       testSandbox.replace(
         testConnectionController,
         '_getConnectionInfoWithSecrets',
-        () =>
+        (connectionInfo) =>
           Promise.resolve({
-            ...savedConnection,
+            ...connectionInfo,
             secretStorageLocation: SecretStorageLocation.Keytar,
           } as any)
       );
+      const trackStub = testSandbox.stub(testTelemetryService, 'track');
 
-      const fakeTrack = sinon.stub();
-      testSandbox.replace(testTelemetryService, 'track', fakeTrack);
-
+      // Load the connections
```

Comment:
The comment is misleading since below with clearAllConnections we clean connections.

Reply:
Sorry for missing that, updating it.